src/types/list.rs

```diff
@@ -1054,9 +1059,13 @@ mod tests {
             }
         }
 
-        impl ToPyObject for Bad {
-            fn to_object(&self, py: Python<'_>) -> Py<PyAny> {
-                self.to_owned().into_py(py)
+        impl<'py> IntoPyObject<'py> for Bad {
+            type Target = PyInt;
+            type Output = crate::Bound<'py, Self::Target>;
+            type Error = Infallible;
+
+            fn into_pyobject(self, py: Python<'py>) -> Result<Self::Output, Self::Error> {
+                self.clone().0.into_pyobject(py)
```

Comment:
I wonder whether it's simpler to remove `Clone` from the picture and instead just have the panic produced in here?

IIRC, the main point of the test is to check that panics while constructing the list free stuff properly, `Clone` was probably just the bug report which came in or similar.

Reply:
Yes, I think that makes it a bit clearer. I changed this, removed the now unnecessary `cfg` and renamed the test (also for `PyTuple`)